test/software/amazon/ionschema/IonSchemaTestRunner.kt

```diff
@@ -0,0 +1,221 @@
+package software.amazon.ionschema
+
+import org.junit.Assert.*
+import org.junit.runner.notification.RunNotifier
+import org.junit.runner.RunWith
+import org.junit.runners.Suite
+import software.amazon.ion.IonContainer
+import software.amazon.ion.IonList
+import software.amazon.ion.IonSequence
+import software.amazon.ion.IonStruct
+import software.amazon.ion.IonSymbol
+import software.amazon.ion.IonValue
+import software.amazon.ion.system.IonTextWriterBuilder
+import software.amazon.ionschema.internal.SchemaCore
+import software.amazon.ionschema.internal.SchemaImpl
+import software.amazon.ionschema.internal.TypeImpl
+import java.io.File
+import java.io.FileReader
+import java.io.OutputStream
+
+/**
+ * Primary test runner for the file-based test suite.
+ */
+@RunWith(IonSchemaTestRunner::class)
+@Suite.SuiteClasses(IonSchemaTestRunner::class)
+class IonSchemaTestRunner(
+        testClass: Class<Any>
+) : AbstractTestRunner(testClass) {
+
+    private val schemaSystem = IonSchemaSystemBuilder.standard()
+            .withAuthority(AuthorityFilesystem("data/test"))
+            .build()
+
+    private val schemaCore = SchemaCore(schemaSystem)
+
+    private val blacklist = setOf(
+            "data/test/constraints/not/empty_type.isl"
+    )
+
+    private val specialFieldNames = setOf("fields", "element")
+
+    override fun run(notifier: RunNotifier) {
+        val base = "data/test"
+        File(base).walk()
+            .filter { it.isFile }
+            .filter { !it.path.endsWith(".md") }
```

Comment:
Would it be better to filter out anything that doesn't have the `.isl` file extension? Currently that's just `README.md`, but you might have other files lying around in the future.